Add IQS035 rule (#2400)
home-assistant/developers.home-assistant: docs/core/integration-quality-scale/rules/iqs035.md

---

> title: "IQS035 - Actions are added in async_setup"

Reviewer: ```suggestion
title: "IQS035 - Service actions are registered in async_setup"
```

---

IQS035 is as follows:

Actions are added in async_setup

## Reasoning

Integrations can add their own service actions to Home Assistant.

> In the past, they have been frequently added in the `async_setup_entry` method and unloaded in the `async_unload_entry` method.

Reviewer: ```suggestion
In the past, they have been frequently registered in the `async_setup_entry` method and removed in the `async_unload_entry` method.
```

The result of this is that the service actions are only available when there is a loaded entry.

> This is not ideal, since this way we can't validate automations users create that use these service actions, since it is possible that the config entry could not be loaded.

Reviewer: Should we call it configuration entry everywhere? I don't remember how consistent we've been in the other rules.

Author: I'll write down that I will do a search afterwards

We rather prefer integrations to set up their service actions in the `async_setup` method.

> The fact still exist that you need to have at least one loaded entry to have a valid target, but now we can let the user know why the service action did not work.

Reviewer: ```suggestion
This way we can let the user know why the service action did not work, if the targeted configuration entry is not loaded.
```
